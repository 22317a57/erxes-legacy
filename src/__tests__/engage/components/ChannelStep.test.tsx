import { mount, shallow } from 'enzyme';
import ChannelStep from 'modules/engage/components/step/ChannelStep';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('ChannelStep component', () => {
  const defaultProps = {
    onChange: (name: 'method', value: string) => null,
    method: 'method'
  };

  test('renders successfully', () => {
    shallow(<ChannelStep {...defaultProps} />);
  });

  test('renders with default props', () => {
    const control = mount(<ChannelStep {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer
      .create(<ChannelStep {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
