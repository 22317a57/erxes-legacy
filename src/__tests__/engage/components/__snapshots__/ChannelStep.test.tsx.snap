// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelStep component snapshot matches 1`] = `
<div
  className="sc-kkGfuU cTwHKp"
>
  <div
    className="sc-iBmynh GORtw sc-hSdWYo iFcZeJ"
    onClick={[Function]}
    selected={false}
  >
    <i
      className="icon-email-1 sc-bwzfXH hpxbGw"
      icon="email-1"
    />
    <span>
      email
    </span>
    <p>
      Delivered to a user's email inbox Customize with your own templates
    </p>
  </div>
  <div
    className="sc-iBmynh GORtw sc-hSdWYo iFcZeJ"
    onClick={[Function]}
    selected={false}
  >
    <i
      className="icon-chat sc-bwzfXH hpxbGw"
      icon="chat"
    />
    <span>
      messenger
    </span>
    <p>
      Delivered inside your app Reach active users
    </p>
  </div>
</div>
`;
